website/blog/2022-04-14-add-ci-cd-to-bitbucket.md

```diff
@@ -0,0 +1,289 @@
+
+Finally, we need to setup the environment so that all the steps requiring authentication can succeed, this includes:
+
+1. Database authentication - for dbt communicating with the warehouse
+2. Bitbucket Downloads authentication - for storing our dbt artifacts
+
+As with the earlier steps, these specific instructions are for Bitbucket but the basic principles apply to any other platform.
+
+### Database authentication
+
+1. Determine the values of all of the variables in `.ci/profiles.yml` (`DB_{CI,PROD}_{HOST,PORT,USER,PWD,DBNAME,SCHEMA}`)
+2. Go to Repository > Repository Settings > Repository Variables in Bitbucket and define them there, making sure to store any confidential values as “Secured”.
+
+![Bitbucket repository variables settings screenshot](/website/static/img/blog/2022-04-14-add-ci-cd-to-bitbucket/2022-04-14-add-ci-cd-to-bitbucket-image-2.png)
+
+### Bitbucket Downloads authentication
+
+1. Go to Personal Settings > App Passwords in Bitbucket and create a Bitbucket App Password with scope `repository:write`.
+2. Go to Repository > Repository Settings > Repository Variables and define the following:
+    1. `BITBUCKET_USERNAME`, which is not your sign-up e-mail, but rather the username found by clicking your avatar in the top left > Personal settings > Account settings page, under Bitbucket Profile Settings.
+    2. `BITBUCKET_APP_PASSWORD`, making sure to store it as “Secured”
+
+![Bitbucket repository app password scope settings screenshot](/website/static/img/blog/2022-04-14-add-ci-cd-to-bitbucket/2022-04-14-add-ci-cd-to-bitbucket-image-1.png)
+
+### Enable Bitbucket Pipelines
+
+Lastly, under Repository > Repository Settings > Pipelines Settings, check “Enable Pipelines”.
+
+![Bitbucket Pipeline Settings enabling screenshot](/website/static/img/blog/2022-04-14-add-ci-cd-to-bitbucket/2022-04-14-add-ci-cd-to-bitbucket-image-3.png)
+
+## Step 4: Test
+
+You’re all done! Now it’s time to test that things work:
+
+1. Push a change to your main branch. This should trigger a Pipeline. Check that it’s successful.
+2. File a Pull Request with a change to a single model / addition of a single test. Check that only that model/test ran.
+
+## Conclusion
+
+It’s important to remember that CI/CD is a convenience, not a panacea. You must still devise the model logic and determine the appropriate tests. Some things it can do, though: catch more mistakes early, make sure that the database always reflects the most up-to-date code, and decrease the friction in collaboration. By automating the steps that should *always* be taken, it frees you up to think about the unusual steps required (e.g., do your changes to [incremental models](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/configuring-incremental-models) require an additional deployment with `--full-refresh`?) and reduces the amount of review that others’ actions necessitate.
+
+Plus, it’s a good time, and it’s fun to watch the test lights turn green. Ding!
```

Comment:
...I really can't think of another way to end this, but dear lord in heaven there must be a serious sentence somewhere out there, right? Alternately, I think this needs a gif to illustrate the "Ding!"

Reply:
Maybe this: https://c.tenor.com/wemr0g_Do8IAAAAC/wild-child-yes.gif

Reply:
hahahah YES there's no turning back now this is The One ✅ .